### ISSUE TITLE
On GPU, jnp.min incorrectly returns 0 when reducing multiple axes over FP16 array

### ISSUE THREAD

**Issue by shacklettbp** [google/jax#18573]
### Description

I have encountered a major bug in the behavior of jnp.min when working with multidimensional fp16 arrays and reducing over multiple axes. Specifically, given an input array > 0, jnp.min returns 0. 
Here is a minimal reproducer:
```
import jax
from jax import random, numpy as jnp

a = jnp.ones((2, 16385, 1), dtype=jnp.float16)

@jax.jit
def reduce_axes12(a):
    return jnp.min(a, axis=(1, 2))

mins = reduce_axes12(a)
print(mins) # Prints [0. 0.], not [1. 1.]
```

Strangely, this error only occurs if all elements in the input array array are > 0 -- if negative values exist they will correctly be returned as the min. I haven't tested other reduce operations.

Other notes:

This behavior only manifests itself when the input array is large enough (or one axis is large enough?): changing 16385 in the above snippet to 16384 does not reproduce the error. Similarly, removing the 3rd dimension and only performing a reduce across axis 1 does not reproduce the error. I was also not able to reproduce the problem for bfloat16 or float32 arrays.

Checking the compiled IR suggests this is probably an XLA bug - I don't see anything wrong here:
```
HloModule jit_reduce_axes12, is_scheduled=true, entry_computation_layout={(bf16[2,1000000,1]{2,1,0})->bf16[2]{0}}, allow_spmd_sharding_propagation_to_output={true}

%region_0.3.clone (Arg_0.0: f32[], Arg_1.0: f32[]) -> f32[] {
  %Arg_1.0 = f32[] parameter(1)
  %Arg_0.0 = f32[] parameter(0)
  ROOT %minimum.1 = f32[] minimum(f32[] %Arg_0.0, f32[] %Arg_1.0), metadata={op_name="jit(reduce_axes12)/jit(main)/reduce_min[axes=(1, 2)]" source_file="/home/bps/mw/jax_bug.py" source_line=9}
}

%fused_computation (param_0.3: bf16[2,1000000,1]) -> bf16[2] {
  %param_0.3 = bf16[2,1000000,1]{2,1,0} parameter(0)
  %bitcast.9 = bf16[2,1000000]{1,0} bitcast(bf16[2,1000000,1]{2,1,0} %param_0.3)
  %convert.14 = f32[2,1000000]{1,0} convert(bf16[2,1000000]{1,0} %bitcast.9)
  %constant_0 = bf16[] constant(inf)
  %convert.13 = f32[] convert(bf16[] %constant_0)
  %reduce.1 = f32[2]{0} reduce(f32[2,1000000]{1,0} %convert.14, f32[] %convert.13), dimensions={1}, to_apply=%region_0.3.clone, metadata={op_name="jit(reduce_axes12)/jit(main)/reduce_min[axes=(1, 2)]" source_file="/home/bps/mw/jax_bug.py" source_line=9}
  ROOT %convert.12 = bf16[2]{0} convert(f32[2]{0} %reduce.1)
}

ENTRY %main.8 (Arg_0.1: bf16[2,1000000,1]) -> bf16[2] {
  %Arg_0.1 = bf16[2,1000000,1]{2,1,0} parameter(0), sharding={replicated}
  ROOT %fusion = bf16[2]{0} fusion(bf16[2,1000000,1]{2,1,0} %Arg_0.1), kind=kLoop, calls=%fused_computation, frontend_attributes={fingerprint_before_lhs="e10df115962e216d28b10c1b752e58d4"}
}
```

### What jax/jaxlib version are you using?

jax 0.4.20, jaxlib 0.4.20

### Which accelerator(s) are you using?

GPU

### Additional system info

Python 3.10, Linux

### NVIDIA GPU info

```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.29.02              Driver Version: 545.29.02    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3090        Off | 00000000:01:00.0  On |                  N/A |
| 30%   43C    P8              40W / 350W |    812MiB / 24576MiB |      8%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
```

**Comment by hawkinsp** [comment]
I agree. Filed https://github.com/openxla/xla/issues/7112
